syntax = "proto3";

package todoPackage;


//
service Todo {
   // create a todo
  rpc CreateTodo(TodoItem) returns (TodoItem) {}
  //read to do
  rpc ReadTodos(TodoRequestNoParam) returns (TodoItems) {}

}


// void data type
message TodoRequestNoParam {}



message TodoItem {
  int32 id =  1;
  string text = 2;
}

message TodoItems {
  repeated TodoItem items = 1;
}